### ISSUE TITLE
Suggestion: dont "test" all cameras

### ISSUE THREAD

**Issue by rodrigograca31** [steveseguin/vdo.ninja#1098]
On mobile when I click "share your camera" it loads my front facing camera for half a second and then loads the back camera which is the one I want to use.

This makes it feel slow and sluggish. 
it takes literally 5+ seconds from clicking the button to having the desired camera ready to click "START"

**Comment by steveseguin** [comment]
Thank you for the feedback.

The system doesn't quite test all cameras, but it may load the default camera, and then switch to the desired camera after that. 

This was mainly to address some compatibility issues with some devices, settings, and browsers. 

It is definitely is sluggish on mobile though, largely because those cameras can't be operated at the same time, and they take time to start and properly close.

I didn't want to mess with it much as the workflow has proven itself fairly reliable, but its perhaps time I take a second stab at it. I'll review the logic and will attempt to skip the initial default camera in the scenarios that I can

**Comment by steveseguin** [comment]
done.

I've pushed the requested changes for this to [https://vdo.ninja/alpha/](https://vdo.ninja/alpha/) and github, which should achieve faster loading camera times when the desired camera is not the default device.

added notes;

It will still be slow if loading VDO.Ninja for the first time, where permissions are not yet granted. I cannot change cameras until I know the names of the media devices, and that requires loading the default camera for such permissions.

In other cases where permissions are already grant, assuming a modern browser, it will skip that step and load the requested device now directly. This is the main change.

I could perhaps make things slightly faster, by loading the audio and the video tracks at the same time instead of individually, but doing so has in the past has caused conflicts with certain devices and drivers.

Firefox will be rather annoying on first-load still, asking for camera permissions potentially twice, but it shouldn't load slow at least.

Please let me know if you have any issues; these changes will go into production once VDO.Ninja v24 is released officially.

**Comment by rodrigograca31** [comment]
Yeah! went down to half of the time! about 3 seconds now! :partying_face: 
thanks for your work! :heart:
